subsys/net/ip/tcp.c

```diff
@@ -352,6 +348,32 @@ static void tcp_send(struct net_pkt *pkt)
 	tcp_pkt_unref(pkt);
 }
 
+static int tcp_derive_rto(struct tcp *conn)
```

Comment:
What about the initial send in `tcp_send_queued_data()`, shouldn't the retransmission timeout be also randomized in there (i. e. by using `tcp_derive_rto()`).

Reply:
Every time the system goes from resend to normal send mode, the rto_random is recomputed. This way it is only recomputed when the timeout has actually been elapsed rather then every time of arming the timeout timer.

Reply:
Hmm, I'm not sure I follow, perhaps I wasn't clear enough. I meant that this line still uses `tcp_rto` variable as a timeout instead of `tcp_derive_rto()` - so the initial retransmission will always take place after a fixed time regardless of the CONFIG_NET_TCP_RANDOMIZED_RTO:
https://github.com/zephyrproject-rtos/zephyr/blob/0aae1bff9c90a94913a5a0be72131566179eec07/subsys/net/ip/tcp.c#L1207
Unfotunately GH does not allow to comment on that very line (as it's not chagned).

Reply:
As suggested, stored the timeout in the connection instead of a random value, the function tcp_derive_rto has been given the task to compute this value.